flybywiresim/a32nx
# feat(mcdu): Implementation of MCDU airport button

Addresses some of the issues in #1948

## Summary of Changes

Cycles between airports when pressing the airport button on the MCDU.

- Pressing the airport button from any other page, brings up the FPLN page, showing the DEST airport at key 4L
- If in PREFLIGHT phase, subsequent presses cycles between DEST/ORIGIN airports
- If not in PREFLIGHT phase, pressing the button simply shows the DEST airport

## Screenshots (if necessary)


![In-sim, MCDU Airport button](https://user-images.githubusercontent.com/6499570/128491130-d001f02a-5dad-41f3-b412-e3a2a4709ef8.gif)

## References

Primarily, references was taken from issue #1948 

Additionally this video was used to further confirm some of the behavior.
https://youtu.be/EoJ9uqMYH_k?t=195
![MCDU - Airport button](https://user-images.githubusercontent.com/6499570/128424352-902a70da-0c1f-4769-8388-68242c8ab9a4.gif)





## Additional context

I have not taken ALTN airport into account, as it depends on other systems not yet implemented.


Discord username (if different from GitHub): Fibre#0690

## Testing instructions

1. Load up the aircraft.
2. Press the airport button to display the DEST airport.
3. Press the airport button to cycle between DEST/ORIGIN airports.
4. Takeoff
5. Load any other MCDU page than FPLN
6. Press AIRPORT button to ensure FPNL page with DEST airport shown at 4L


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
